integration-tests/src/tests/client/features/restrict_tla.rs

```diff
@@ -0,0 +1,59 @@
+use super::super::process_blocks::create_account;
+use crate::tests::client::utils::TestEnvNightshadeSetupExt;
+use near_chain::ChainGenesis;
+use near_chain_configs::Genesis;
+use near_client::test_utils::TestEnv;
+use near_primitives::errors::{ActionError, ActionErrorKind};
+use near_primitives::types::{AccountId, BlockHeight};
+use near_primitives::views::FinalExecutionStatus;
+use near_primitives_core::version::PROTOCOL_VERSION;
+use nearcore::config::GenesisExt;
+
+#[test]
+fn test_create_top_level_accounts() {
+    let epoch_length: BlockHeight = 5;
+    let account: AccountId = "test0".parse().unwrap();
+    let mut genesis = Genesis::test(vec![account.clone()], 1);
+    genesis.config.epoch_length = epoch_length;
+    genesis.config.protocol_version = PROTOCOL_VERSION;
+    let runtime_config = near_primitives::runtime::config_store::RuntimeConfigStore::new(None);
+    let mut env = TestEnv::builder(ChainGenesis::new(&genesis))
+        .real_epoch_managers(&genesis.config)
+        .nightshade_runtimes_with_runtime_config_store(&genesis, vec![runtime_config])
+        .build();
+
+    let top_level_accounts = [
+        "0x06012c8cf97bead5deae237070f9587f8e7a266d",
+        "0x5e97870f263700f46aa00d967821199b9bc5a120",
+        "0x0000000000000000000000000000000000000000",
+        "alice",
+        "thisisaveryverylongtoplevelaccount",
+    ];
+    for (index, id) in top_level_accounts.iter().enumerate() {
+        let new_account_id = id.parse::<AccountId>().unwrap();
+        let tx_hash = create_account(
+            &mut env,
+            account.clone(),
+            new_account_id.clone(),
+            epoch_length,
+            1 + index as u64 * epoch_length,
+            PROTOCOL_VERSION,
+        );
+        let transaction_result =
+            env.clients[0].chain.get_final_transaction_result(&tx_hash).unwrap();
+        assert_eq!(
+            transaction_result.status,
+            FinalExecutionStatus::Failure(
```

Comment:
I'm a bit confused that all of the accounts failed to be created. Are the hex accounts above not implicit accounts? Can you add a comment or adjust the name of the test to reflect that it's testing this NEP specifically?